### DISCUSSION TITLE
Make calendar layout allday row working with datetime fields

### DISCUSSION THREAD

**Discussion by katywings** [directus/directus#15784] [Feature Requests]
Currently the calendar layout's allday column only works for collections were the end-date is specifically a date without time. What do you think about making the functionality work with datetime end-date fields?

The specific logic is here: https://github.com/directus/directus/blob/3a614d0058172987fd2ecdfc58e80b1a55db7863/app/src/layouts/calendar/index.ts#L295

![](https://user-images.githubusercontent.com/4012401/192772812-000ca694-0ce8-4e25-ab74-26ab09d2d810.png)

My suggestion is changing the logic to something like this:
```tsx
let endDate: string | undefined = endDateField.value && item[endDateField.value];

let allDay = false;

if (endDateFieldInfo.value && endDate) {
  const fieldType = endDateFieldInfo.value.type;
  const isDateTime = fieldType === 'dateTime';
  const dateFormat = isDateTime ? "yyyy-MM-dd'T'HH:mm:ss": 'yyyy-MM-dd';
  const date = parse(endDate, dateFormat, new Date());
  if (format(date, 'H:m:s') == '0:0:0') {
    allDay = true;
    // FullCalendar uses exclusive end moments, so we'll have to increment the end date by 1 to get the
    // expected result in the calendar
    date.setDate(date.getDate() + 1);
    endDate = format(date, dateFormat);
  }				
}
```

This way all events ending at 0:00 would be shown on the allday row.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
